packages-exp/messaging-exp/src/helpers/logToFirelog.ts

```diff
@@ -98,10 +103,21 @@ export async function _dispatchLogEvents(
           }
         );
 
-        break;
+        if (response.ok) {
+          // existing the do-while interactive retry logic because a 200 response code has been
+          // returned.
+          break;
+        }
+
+        if (!response.ok) {
+          throw new Error(
+            'Non-200 code is returned in fetch to Firelog endpoint.'
+          );
```

Comment:
You probably don't want to retry unrecoverable error codes, such as 400 and 401. Remote config only retries for certain error codes - https://github.com/firebase/firebase-js-sdk/blob/master/packages/remote-config/src/client/retrying_client.ts#L68-L81